test/unit/test_visual_recognition_v4.py

```diff
@@ -12,824 +13,1199 @@
-                                "width": 5,
-                                "height": 2
-                            },
-                            "object": "object"
-                        }]
-                    }]
-                },
-                "source": {
-                    "archive_filename": "archive_filename",
-                    "filename": "filename",
-                    "type": "file",
-                    "resolved_url": "resolved_url",
-                    "source_url": "source_url"
-                },
-                "errors": {
-                    "code":
-                        "invalid_field",
-                    "message":
-                        "The date provided for `version` is not valid. Specify dates in `YYYY-MM-DD` format.",
-                    "more_info":
-                        "https://cloud.ibm.com/apidocs/visual-recognition-v4#versioning",
-                    "target": {
-                        "type": "parameter",
-                        "name": "version"
-                    }
-                },
-                "dimensions": {
-                    "width": 6,
-                    "height": 0
-                }
-            }],
-            "trace":
-                "trace",
-            "warnings": [{
-                "code": "invalid_field",
-                "more_info": "more_info",
-                "message": "message"
-            }, {
-                "code": "invalid_field",
-                "more_info": "more_info",
-                "message": "message"
-            }]
-        }
+        return url
+
+    def add_mock_response(self, url, response):
         responses.add(responses.POST,
                       url,
                       body=json.dumps(response),
                       status=200,
                       content_type='application/json')
 
-        authenticator = IAMAuthenticator('bogusapikey')
-        service = ibm_watson.VisualRecognitionV4('YYYY-MM-DD',
-                                                 authenticator=authenticator)
+    def call_service(self, body):
+        service = VisualRecognitionV4(authenticator=NoAuthAuthenticator(),
+                                      version='2019-02-11')
         service.set_service_url(base_url)
+        output = service.analyze(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body['collection_ids'] = ['collection_id1, collection_id2']
+        body['features'] = ['test']
+        body['image_url'] = ['https://upload.wikimedia.org/wikipedia/commons/thumb/4/47/American_Eskimo_Dog.jpg/1280px-American_Eskimo_Dog.jpg']
+        body['threshold'] = 12345.0
+        body['images_file'] = [FileWithMetadata(tempfile.NamedTemporaryFile())]
+        return body
+
+    def construct_required_body(self):
+        body = dict()
+        body['collection_ids'] = ['fake']
+        body['features'] = [AnalyzeEnums.Features.OBJECTS.value]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['AnalyzeEnums' may be undefined, or defined from star imports: ibm_watson.visual_recognition_v4](https://app.codacy.com/app/watson-developer-cloud/python-sdk/pullRequest?prid=4546453)